src/nvim/api/buffer.c

```diff
@@ -334,6 +347,13 @@ void nvim_buf_set_lines(uint64_t channel_id, Buffer buffer, Integer start, Integ
     return;
   }
 
+  // load buffer first if it's not loaded
+  if (!buf_ensure_loaded(buf)) {
+    api_set_error(err, kErrorTypeException, "Failed to load buffer");
+    return;
+  }
+
+  // would removing the following break something?
   // load buffer first if it's not loaded
   if (buf->b_ml.ml_mfp == NULL) {
```

Comment:
not sure if removing the following would have some side effects

Reply:
This is redundant with your addition and can be savely removed :)  (the `buf->b_ml.ml_mfp == NULL` check was not needed)